src/pages-and-resources/discussions/app-config-form/apps/legacy/LegacyConfigForm.jsx

```diff
@@ -123,40 +123,10 @@ function LegacyConfigForm({
 }
 
 LegacyConfigForm.propTypes = {
-  appConfig: PropTypes.shape({
-    divideByCohorts: PropTypes.bool.isRequired,
-    divideCourseTopicsByCohorts: PropTypes.bool.isRequired,
-    allowAnonymousPosts: PropTypes.bool.isRequired,
-    allowAnonymousPostsPeers: PropTypes.bool.isRequired,
-    blackoutDates: PropTypes.arrayOf(PropTypes.shape({
-      id: PropTypes.string,
-      startDate: PropTypes.string,
-      endDate: PropTypes.string,
-      startTime: PropTypes.string,
-      endTime: PropTypes.string,
-      status: PropTypes.string,
-    })),
-    discussionTopics: PropTypes.arrayOf(PropTypes.shape({
-      name: PropTypes.string,
-      id: PropTypes.string,
-    })),
-  }),
   onSubmit: PropTypes.func.isRequired,
   // eslint-disable-next-line react/forbid-prop-types
   formRef: PropTypes.object.isRequired,
   intl: intlShape.isRequired,
-  title: PropTypes.string.isRequired,
-};
-
-LegacyConfigForm.defaultProps = {
-  appConfig: {
-    divideByCohorts: false,
-    allowDivisionByUnit: false,
-    divideCourseTopicsByCohorts: false,
-    allowAnonymousPosts: false,
-    allowAnonymousPostsPeers: false,
-    blackoutDates: [],
```

Comment:
What if, now, these config keys are None? is it possible to have non-default values returned from the API when these values are not set in the course?

Reply:
We are creating a default object at the start of the component. So these keys are not None.
 `const LegacyAppConfig = {
    ...appConfig,
    allowAnonymousPosts: appConfig.allowAnonymousPosts || false,
    allowAnonymousPostsPeers: appConfig.allowAnonymousPostsPeers || false,
    blackoutDates: appConfig.blackoutDates || [],
    discussionTopics: appConfig.discussionTopics || [],
    divideByCohorts: appConfig.divideByCohorts || false,
    divideCourseTopicsByCohorts`